### PR TITLE
feat(staking): Add staking actions

### PR DESCRIPTION
**Pull request by Hemanthghs**
Part of #900 

## Summary by CodeRabbit

- **New Features**
  - Search functionality with optional query parameters added to the staking interface.
  - New components for active and inactive validators, including pagination and filtering based on search terms.
  - Delegation, undelegation, and redelegation dialog components added for transaction operations.
  - Staking actions menu updated to handle new delegation-related actions.

- **Enhancements**
  - Staking page now accepts additional parameters for improved transaction handling.
  - Staking sidebar updated with new transaction functions and improved UI for claiming rewards and staking.
  - Validator components updated to display detailed information and enable delegation actions.

- **Bug Fixes**
  - Fixed issues with the display of available balance and parsing of denomination amounts in staking components.
  - Addressed UI inconsistencies in staking cards and dialogs.

- **Refactor**
  - Consolidated imports and streamlined component signatures across staking-related files.
  - Enhanced state management and event handling in staking components for better user experience.

- **Style**
  - Updated CSS styles for staking components to improve layout and visual appeal.
  - Introduced new styles for pagination components and action buttons.

- **Documentation**
  - Added detailed comments and summaries to staking components for better code understanding and maintenance.

- **Chores**
  - Removed unused imports and code segments to clean up the codebase.
  - Updated constants and utility functions to support new staking features.

### CODE REVIEW

**Comment by Hemanthghs** [comment]
@coderrabbitai review

**Comment by Hemanthghs** [comment]
@coderabbitai review

**Comment by aleem1314** [comment]
@Hemanthghs can you address above review comments? Most of these are related to typos and small improvements

**Comment by Hemanthghs** [comment]
> @Hemanthghs can you address above review comments? Most of these are related to typos and small improvements

Okay @aleem1314

File: frontend/src/app/(routes)/staking/components/StakingCard.tsx [hunk 1]
```diff
@@ -18,16 +30,10 @@ const StakingCard = ({
...
-  coinDenom: string;
-  rewards: number;
-}) => {
+  onMenuAction,
+  validatorInfo,
+  chainID,
+}: StakingCardProps) => {
   const [isMenuOpen, setMenuOpen] = useState(false);
   const menuRef = useRef<HTMLDivElement>(null);
   const menuRef2 = useRef<HTMLDivElement>(null);
```
[the remark this answers is not in the document]
Comment (Hemanthghs) [reply frontend/src/app/(routes)/staking/components/StakingCard.tsx:39]: Both are for different purpose

**Pull request merged**